### ISSUE TITLE
hardmode recipes for microminers and other mods are not gregged enough.

### ISSUE THREAD

**Issue by lyeochan** [Nomi-CEu/nomi-ceu#341]
**Is your feature request related to a problem? Please describe.**
Not a problem on normal mode, but on hardmode I was expecting more recipes changes especially on the core feature of the pack: microminers (check the recipe for the basic mining laser [that recipe looks like its from a mc creator "laser mod"],  another recipe I found lacking was the reinforced mining laser[this one at least adds some GT circuitry but redstone blocks + glass and netherquartz blocks are not what would you call a gregtech recipe] . 

This problem is also present in some other mods (like draconic evolution, draconium is a multistep material that you unlock towards the endgame, however recipes that use draconium are too basic, 4 draconium ingot + 4 blaze powder + chorus plant for a dislocator, and then the advanced dislocator just uses more draconium ingot, a couple enderpears and a dragon egg? it feels such a whiplash to go from making advanced chemicals and processes to suddenly using a ingot in a recipe (especially cause GT almost never uses ingots directly to make machines or items, and blaze powder).

The pack does already have lots of cool "greggified" recipes, the excitation coil, the canola press, war cores, etc, and in my option could be even better with more "greggified" mods.

**Describe the solution you'd like**
Recipes that include materials that you have to process in gregtech machines.
The basic mining laser for example could use: Fused quartz instead of glass, a red lens or a ruby lens instead of the redstone block and some lv emmiters.

I am willing to make those recipes if so desired.

**Describe how the solution will work**
hardmode will properly feel like a hardmode, a propper rework on costs of microminers will make sure players invest in their infrastructure and feel the weight of making the first few microminers. it will lot more rewarding automating a even more complex chain of microminer parts to finally get passive resource income.

**Describe alternatives you've considered**
Alongside the example I gave on the basic mining laser, lots of places where ingots are used they could be switched to plates, and where ender pearls are found in recipes they could be switched to a tier of field generator or some other more complex material. 
Opinium Cores and Crystaltine ingots both feel extremelly out of place in a gregtech context [look a their recipes, it looks like just a vomit of vanilla materials, and what I would describe as a """""""""""hard"""""""""" craft]

**Additional context**
Here are some recipes I found, personally, lacking, and feel like  could have some improvement.

![image](https://user-images.githubusercontent.com/94973750/234142996-0e79fa9d-e3d7-45f2-9b03-b3a1c23be3fc.png)
![image](https://user-images.githubusercontent.com/94973750/234143009-f19f484d-5792-4a16-a29e-bd2858291c72.png)
![image](https://user-images.githubusercontent.com/94973750/234142923-d3880d92-58be-40b5-bff9-a34171ded977.png)
![image](https://user-images.githubusercontent.com/94973750/234142954-9801d633-a65b-4bdb-a85c-ccc30021af61.png)

**Comment by Brownbeltninja** [comment]
These are some good points, I think that the mining lasers at least shouldn't be made of pure just glass (perhaps tempered or higher tiers of glass?) and the materials for it should be non vanilla, the redstone could be a red lens, this will encourage players to automate chemical dyes and the quartz could probably be subsituted for at least some form of quartz plates, the recipe for it is just a bit basic.

**Comment by lyeochan** [comment]
well, the other parts of the microminers are do use GT components, so yeah its probably enough gt component crafting (unless said other components are made cheaper).

i think the whole point could be summed up to: Add more non vanilla materials to crafts, ofc depends on each craft and at what tier u make said craft!

**Comment by Folanlron** [comment]
anything like this would probably be after 1.7.. 

This would also have to be a complete overhaul/re-balance of the MM system for just HM..

**Comment by Folanlron** [comment]
grr Github stuff is so weird..


Anything like this would have to be after 1.7..

As this would be a complete rebalance of the MM system for HM

**Comment by lyeochan** [comment]
I understand this isn't a priority, however, i don't think it would be as deep as a complete rebalance, slightly more involved recipes that uses gt materials don't necessarily make it much more expensive (especially on the lower tiers), either way I am looking forward to see how the pack's HM is further improved. If assistance is required for stuff like this, I am willing to help, as I said. 

Good luck with the pack dev!

**Comment by IntegerLimit** [comment]
Yes, as Folanlron said, if this happens, it would be on 1.7. Especially, if we want different modes to have different drac evolution fusion recipes, that would be a pain with the current setup, so it would be better to be done with GS.

**Comment by IntegerLimit** [comment]
I've added this to be looked at, in the ever increasing 1.7 milestone.

**Comment by lyeochan** [comment]
lel, here the one thing u told me to put here:

change extended crafting recipes to ass recipes and assline recipes!
maybe extended crafting -> assembly machines & lines

**Comment by lyeochan** [comment]
oh here, i remember now , recipes IMO could do with gregness, and a possible gratification for them:
iron door -> same structure as wood door, but with iron plates
saddle -> remove thermal foundation recipe
Advanced inscribed, wireless connector, spatial IO port, cell workbench, io port, wireless booster -> stainless steel plates / dark steel plates instead of iron to fit the rest of the tweaks.
remove the shaped crafting of spatial storage units, instead keep only the shapeless since the shaped one bypasses gregness
toggle bus -> Redstone plate instead of redstone dust
--------
Actually additions:
For the most part its very well gregged IMO, however id change the base material for some of the crystals (namely for enori, and void. Something White like quartz and obsidian might fit better?)
some stuff could do with red alloy plates / wires instead of redstone tho
shapeless recipe to convert fluid plaer and picker
auto breaker / placer  don't make much sense being made out of cobble (at least not here IMO, although not a egregious recipe) -> steel instead would fit IMO
energy laser relay -> dark steel or steel instead of obi.
phantom connector -> could do with a circuit (even LV one), very powerful tool!
-----------
AR, Greggers!
-----------
armor plus
most of the recipes are just vanilla endgame materials vomited into a large crafting table (or mob drops) , most of them are irrelevant, and poses no challenge to craft at all, especially at lategame.
-----------
Buiding gadgets, change to iron plates, cause whynot, but id also remove the energy costs for them, let players build as soon as possible, lots of ppl basebuild as a way to not get overburnt on the earlygame, diving them easy access to powerful building tools could make the process a lot more enjoyable.
----------
Content tweaker
Basic mining laser for MMs: red lends instead of redstone block, fused quartz instead of glass, and electrum plates instead of quartz (or quartz plates)
Reinforced Mining laser: red lends instead of redstone block, any higher tier glass instead of vanilla glass, emmiters instead of quartz blocks, electrum instead of nether quarts.
Microverse casings should follow casings craft like the other GT multiblock casings?
---------
dark utilities, is this mod even useful on HM, is this mod even usefull?
well either way yeet some void crystals in the recipes and should be fine, or prune it down a lot
---------
Draconic
particle generator -> could do with a GT emmiter, and fluxed electrum instead of blaze rods
energy infuser , Energy core, Celestial Manip -> valid for many Draconic crafts, use Draconium plates instead of ingots, its more "greggy"
Dislocator receptable -> Iron plates at minimum, but given the color, electrum or fluxed electrum could be fine
entity detector / adv entity detector , energy pylon-> recipe too vanilla-ish, some AA crystals / empowered crystals IMO could do fine here, same issue with draconium plates.
Energy  Relay crystals, wireless energy crystals -> emmiters, receivers or even field gens instead of the simple eyes of ender could make it gregorical enough.
Crafting injectors -> Netherstar simulator, but egh, fine ig?
Draconic Core -> Plates instead of ingots, or even rods
Other Cores -> other cores feel fine, imo
Upgrades -> AA crystals, and plates, cant go wrong with that.
Dislocator and adv dislocator -> Fiel generator,instead of chorus, draconium plates instead of dust, and maybe fluxed electrum plates too?, for the advanced one, substitute ender pearls for enderium.
--------
EIO
Plates/rods instead of ingots most of the time this is enough to get something to be greggy.
-------
Extended Crafting
Crystaltine ingots are cancer xd, no other way to describe this and too similar to avaritia's diamond matrix ingot
-------
Extra Utils
for the most part, good, however opinium cores are one of the most cancerous craft in the world, its the definition of """""""""""""hard"""""""""""""" in expert packs and vanilla vomit crafts (project E packs)
---------
gregtech, gregtech
--------
Nuclearcraft
plates instead of ingot
some blocks needs gratification, like the machine interfaces (which uses a non gt motor), 
---------
Packaged auto
packaged extender needs the greg power
---------
Simple jetpacks
Plate issue, mostly fine
--------
Solar Flux, I hate solar flux, however the recipes don't seem that dumb, just plate issue
--------
Thermal Expansions
Plate Issue
Some items like Flux Capacitor could also do with using redstone alloy plates, along their plate issue
Microminer Engine Core / Energy Cells -> could use liquid redstone on canning machine?, or maybe liquid empowered restonia on the case of the more advanced crafts
Add a inefficient plate bending machine recipe for mana infused ingots to be able to fix the thermal device casing plate issue
 
 
 ------
 This is all I could find on by scouring JEI in a couple minutes.

**Comment by lyeochan** [comment]
i freaking hate how github works sometimes ergh

**Comment by IntegerLimit** [comment]
Wow, thanks for the detailed text. Those sound like some good changes, but still, this will be made in 1.7, as before that, those will be too big changes.

**Comment by lyeochan** [comment]
still, if you want i can make some scripts, and even playtest with the extra difficult stuff.

lots of stuff on this problem is an easy 1 line fix per item like:

//recipes.replaceAllOccurences(IIngredient toReplace, IIngredient replaceWith, @Optional IIngredient forOutput);
//Only replaces in recipes with a diamond_sword as output
recipes.replaceAllOccurences(<ore:gemDiamond>, <ore:blockDiamond>, <minecraft:diamond_sword>);

**Comment by IntegerLimit** [comment]
I would rather not, as this would have to be re-implemented into GroovyScript. However, if you can instead make a PR to the groovy branch, written in groovy, that would be fine.

**Comment by lyeochan** [comment]
ah understandable, i'll see if i cna learn groove script in time!
shouldn't be too hard, especially if I just copy from other examples.

**Comment by IntegerLimit** [comment]
Check out the groovy branch!

**Comment by Folanlron** [comment]
> Advanced inscribed, wireless connector, spatial IO port, cell workbench, io port, wireless booster -> stainless steel plates / dark steel plates instead of iron to fit the rest of the tweaks. 

No, AE is suppose to stay at MV for a reason, already having blue steel for the Inscriber is fine, All the Spatial IO isn't used for anything at all(And honestly no one uses it anyway)other then for Naq Reactor(If you are using Naq reactors on HM you got other problems), IO Port should stay were it is as starting out you are flipping drives alot, adjusting these items to another tier would ruin a lot of progression with-in AE its self.


> toggle bus -> Redstone plate instead of redstone dust - Nobody uses it because its garbage.




> auto breaker / placer don't make much sense being made out of cobble (at least not here IMO, although not a egregious recipe) -> steel instead would fit IMO, 

The Auto-breaker is this way cause Sugar Cane farms are so important at the start, even in HM..

> ## AR, Greggers!
> ## armor plus
> most of the recipes are just vanilla endgame materials vomited into a large crafting table (or mob drops) , most of them are irrelevant, and poses no challenge to craft at all, especially at lategame.

HM All the armor is useless and is only good for the final crafting Items, Why even look at them?



> Microverse casings should follow casings craft like the other GT multiblock casings? 

Casings do Follow the same idea, 8 ingots per 2 casings, just cause there is no lathe recipe doesn't mean the recipe isn't adjusted for it.

> ## dark utilities, is this mod even useful on HM, is this mod even usefull?

Vector Plates needed for Growth Medium(Chicken Farm), aka WetWare Circuits


> ## Extended Crafting
> Crystaltine ingots are cancer xd, no other way to describe this and too similar to avaritia's diamond matrix ingot 


HM has a alternate recipe for the crystal ingots, yes the crystalizer is expensive as it should be

> ## Extra Utils
> for the most part, good, however opinium cores are one of the most cancerous craft in the world, its the definition of """""""""""""hard"""""""""""""" in expert packs and vanilla vomit crafts (project E packs)

I don't see the point of changing these, The Cores are only used for 2 items, and 1 of them is useless in this pack, the XU2 RF Source is sort of useful.. but GT Energy Emitter is better.


> ## Packaged auto
> packaged extender needs the greg power

Why even waste the time to do this? all this would do is cause 1 less crafting table in Extended Crafting Clusters, AE power is fine for this.
